src/v/storage/disk_log_impl.cc

```diff
@@ -88,7 +88,16 @@ bool deletion_exempt(const model::ntp& ntp) {
                                  || ntp.ns() == model::kafka_internal_namespace;
     bool is_tx_manager_ntp = ntp.ns == model::kafka_internal_namespace
                              && ntp.tp.topic == model::tx_manager_topic;
-    return !is_tx_manager_ntp && is_internal_namespace;
+    bool is_consumer_offsets_ntp = ntp.ns()
+                                     == model::kafka_consumer_offsets_nt.ns()
+                                   && ntp.tp.topic
+                                        == model::kafka_consumer_offsets_nt.tp;
+    bool is_schema_registry_ntp
+      = ntp.ns() == model::kafka_internal_namespace
+        && ntp.tp.topic == model::schema_registry_internal_tp.topic;
+
+    return is_tx_manager_ntp || is_consumer_offsets_ntp
```

Comment:
Looking at the original code, it looks like the tx_manager shouldn't be considered deletion exempt